packages/editor-ui/src/App.vue

```diff
@@ -110,52 +107,45 @@ export default mixins(
 				return;
 			}
 
-			if (this.$route.name === 'SetupView' && !this.isUserManagementEnabled) {
-				this.$router.replace('/');
-
-				this.loading = false;
-				return;
-			}
-
 			if (this.canCurrentUserAccessView(this.$router.currentRoute.name)) {
-				this.loading = false;
-
 				return;
 			}
 
+			// if cannot access page and not logged in, ask to sign in
 			const user = this.currentUser as IUser | null;
 			if (!user) {
 				const redirect =
 					this.$route.query.redirect ||
 					encodeURIComponent(`${window.location.pathname}${window.location.search}`);
 				this.$router.replace({ name: 'SigninView', query: { redirect } });
-			} else {
-				if (typeof this.$route.query.redirect === 'string') {
-					const redirect = decodeURIComponent(this.$route.query.redirect);
-					if (redirect.startsWith('/')) {
-						// protect against phishing
-						this.$router.replace(redirect);
-					}
-				} else {
-					this.$router.replace({ name: 'NodeViewNew' });
+				return;
+			}
+
+			// if cannot access page and is logged in, respect signin redirect
+			if (this.$route.name === 'SigninView' && typeof this.$route.query.redirect === 'string') {
+				const redirect = decodeURIComponent(this.$route.query.redirect);
+				if (redirect.startsWith('/')) { // protect against phishing
+					this.$router.replace(redirect);
+					return;
 				}
 			}
 
-			this.loading = false;
+			// if cannot access page and is logged in
+			this.$router.replace({ name: 'Homepage' });
+		},
+		redirectIfNecessary() {
+			const redirect = this.$route.meta && typeof this.$route.meta.getRedirect === 'function' && this.$route.meta.getRedirect(this.$store);
+			if (redirect) {
+				this.$router.replace(redirect);
+			}
 		},
 	},
 	async mounted() {
+		this.logHiringBanner();
```

Comment:
would move this to after `this.authenticate()`

Reply:
it does not matter. mostly I put it here because I want this to take effect before any of the telemetry logs are made